### ISSUE TITLE
Helm 2.10.0 fails to build on macOS

### ISSUE THREAD

**Issue by fxcoudert** [helm/helm#4584]
As part of Homebrew testing for Mojave (macOS 10.14) https://github.com/Homebrew/homebrew-core/issues/28817, we found out that helm 2.10.0 fails to build on that OS version, with Xcode 10:

```
�[0;32m[INFO]	�[m--> Exporting golang.org/x/text
�[0;32m[INFO]	�[mReplacing existing vendor dependencies
�[0;32m[INFO]	�[mRemoving nested vendor and Godeps/_workspace directories...
�[0;32m[INFO]	�[mRemoving: /private/tmp/kubernetes-helm-20180904-47960-12ybd2d/src/k8s.io/helm/vendor/github.com/cyphar/filepath-securejoin/vendor
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x12fca82]

goroutine 1 [running]:
github.com/Masterminds/glide/path.StripVendor.func1(0xc4206e6630, 0x82, 0x0, 0x0, 0x14f11a0, 0xc4201e9b30, 0x0, 0x0)
	/private/tmp/glide-20180816-23735-1ovdnmf/glide-0.13.1/src/github.com/Masterminds/glide/path/strip.go:28 +0x42
path/filepath.walk(0xc4205ef900, 0x77, 0x14f8320, 0xc4209df450, 0xc4203e6040, 0x0, 0x0)
	/usr/local/Cellar/go/1.10.3/libexec/src/path/filepath/path.go:377 +0x20d
path/filepath.walk(0xc420620380, 0x70, 0x14f8320, 0xc4209deea0, 0xc4203e6040, 0x0, 0x0)
	/usr/local/Cellar/go/1.10.3/libexec/src/path/filepath/path.go:381 +0x2c2
path/filepath.walk(0xc4202eeb40, 0x5c, 0x14f8320, 0xc4209dedd0, 0xc4203e6040, 0x0, 0x0)
	/usr/local/Cellar/go/1.10.3/libexec/src/path/filepath/path.go:381 +0x2c2
path/filepath.walk(0xc4202ee360, 0x55, 0x14f8320, 0xc420590dd0, 0xc4203e6040, 0x0, 0x0)
	/usr/local/Cellar/go/1.10.3/libexec/src/path/filepath/path.go:381 +0x2c2
path/filepath.walk(0xc4202e3040, 0x4a, 0x14f8320, 0xc4205211e0, 0xc4203e6040, 0x0, 0x20)
	/usr/local/Cellar/go/1.10.3/libexec/src/path/filepath/path.go:381 +0x2c2
path/filepath.Walk(0xc4202e3040, 0x4a, 0xc4203e6040, 0xc420521110, 0x0)
	/usr/local/Cellar/go/1.10.3/libexec/src/path/filepath/path.go:403 +0x106
github.com/Masterminds/glide/path.StripVendor(0x14ae577, 0x3b)
	/private/tmp/glide-20180816-23735-1ovdnmf/glide-0.13.1/src/github.com/Masterminds/glide/path/strip.go:22 +0x10a
github.com/Masterminds/glide/action.Install(0xc4201e3540, 0x149a201)
	/private/tmp/glide-20180816-23735-1ovdnmf/glide-0.13.1/src/github.com/Masterminds/glide/action/install.go:62 +0x35b
main.commands.func12(0xc42023c420, 0x0, 0xc42023c420)
	/private/tmp/glide-20180816-23735-1ovdnmf/glide-0.13.1/src/github.com/Masterminds/glide/glide.go:510 +0x2bf
github.com/Masterminds/glide/vendor/github.com/codegangsta/cli.HandleAction(0x140f100, 0x14b8b40, 0xc42023c420, 0xc4201f6300, 0x0)
	/private/tmp/glide-20180816-23735-1ovdnmf/glide-0.13.1/src/github.com/Masterminds/glide/vendor/github.com/codegangsta/cli/app.go:490 +0xc8
github.com/Masterminds/glide/vendor/github.com/codegangsta/cli.Command.Run(0x149859e, 0x7, 0x1496c20, 0x1, 0x0, 0x0, 0x0, 0x14a3b3b, 0x20, 0x0, ...)
	/private/tmp/glide-20180816-23735-1ovdnmf/glide-0.13.1/src/github.com/Masterminds/glide/vendor/github.com/codegangsta/cli/command.go:210 +0xa36
github.com/Masterminds/glide/vendor/github.com/codegangsta/cli.(*App).Run(0xc42022a000, 0xc4200a8090, 0x3, 0x3, 0x0, 0x0)
	/private/tmp/glide-20180816-23735-1ovdnmf/glide-0.13.1/src/github.com/Masterminds/glide/vendor/github.com/codegangsta/cli/app.go:255 +0x6a0
main.main()
	/private/tmp/glide-20180816-23735-1ovdnmf/glide-0.13.1/src/github.com/Masterminds/glide/glide.go:104 +0x5a9
make: *** [bootstrap] Error 2
```

(which happens during `make bootstrap`). I see there are other “invalid memory address or nil pointer dereference” issues open, but the backtrace appears different on this one, so I am filing this.

**Comment by henrynash** [comment]
I'll try and reproduce this - however, this happened to me with xCode 9, and the solution was to to a new reinstall glide.

**Comment by fxcoudert** [comment]
I can confirm it on several machines, both locally and on our CI machines. Reinstalling glide is done at each CI cycle, and did not fix the issue.

**Comment by fxcoudert** [comment]
It seems it is actually a `glide` issue: https://github.com/Masterminds/glide/pull/990

**Comment by fxcoudert** [comment]
With Homebrew's glide 0.13.1, we can now see the same build failures on all our macOS machines, from Sierra to Mojave, i.e. with Xcode 9 and 10:

```
�[0;32m[INFO]	�[m--> Exporting k8s.io/apiserver
�[0;32m[INFO]	�[mReplacing existing vendor dependencies
�[0;32m[INFO]	�[mRemoving nested vendor and Godeps/_workspace directories...
�[0;32m[INFO]	�[mRemoving: /private/tmp/kubernetes-helm-20180924-57316-16wn6ua/src/k8s.io/helm/vendor/github.com/cyphar/filepath-securejoin/vendor
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x131f182]

goroutine 1 [running]:
github.com/Masterminds/glide/path.StripVendor.func1(0xc000036090, 0x82, 0x0, 0x0, 0x1540360, 0xc000241200, 0x0, 0x0)
	/private/tmp/glide-20180907-75095-1ck8wl4/glide-0.13.1/src/github.com/Masterminds/glide/path/strip.go:28 +0x42
path/filepath.walk(0xc00073f600, 0x77, 0x1547640, 0xc00057e5b0, 0xc0003e9140, 0x0, 0x0)
	/usr/local/Cellar/go/1.11/libexec/src/path/filepath/path.go:378 +0x21f
path/filepath.walk(0xc0004e3490, 0x70, 0x1547640, 0xc00057e000, 0xc0003e9140, 0x0, 0x0)
	/usr/local/Cellar/go/1.11/libexec/src/path/filepath/path.go:382 +0x2fe
path/filepath.walk(0xc00028c960, 0x5c, 0x1547640, 0xc0004ddee0, 0xc0003e9140, 0x0, 0x0)
	/usr/local/Cellar/go/1.11/libexec/src/path/filepath/path.go:382 +0x2fe
path/filepath.walk(0xc00028c000, 0x55, 0x1547640, 0xc000210c30, 0xc0003e9140, 0x0, 0x0)
	/usr/local/Cellar/go/1.11/libexec/src/path/filepath/path.go:382 +0x2fe
path/filepath.walk(0xc0005f6410, 0x4a, 0x1547640, 0xc0001ba270, 0xc0003e9140, 0x0, 0x20)
	/usr/local/Cellar/go/1.11/libexec/src/path/filepath/path.go:382 +0x2fe
path/filepath.Walk(0xc0005f6410, 0x4a, 0xc0003e9140, 0xc0001ba1a0, 0x0)
	/usr/local/Cellar/go/1.11/libexec/src/path/filepath/path.go:404 +0x105
github.com/Masterminds/glide/path.StripVendor(0x14d5ddf, 0x3b)
	/private/tmp/glide-20180907-75095-1ck8wl4/glide-0.13.1/src/github.com/Masterminds/glide/path/strip.go:22 +0x106
github.com/Masterminds/glide/action.Install(0xc0000a5480, 0x14c1501)
	/private/tmp/glide-20180907-75095-1ck8wl4/glide-0.13.1/src/github.com/Masterminds/glide/action/install.go:62 +0x351
main.commands.func12(0xc0001166e0, 0x0, 0xc0001166e0)
	/private/tmp/glide-20180907-75095-1ck8wl4/glide-0.13.1/src/github.com/Masterminds/glide/glide.go:510 +0x2b5
github.com/Masterminds/glide/vendor/github.com/codegangsta/cli.HandleAction(0x1438080, 0x14e03f0, 0xc0001166e0, 0xc0000b7100, 0x0)
	/private/tmp/glide-20180907-75095-1ck8wl4/glide-0.13.1/src/github.com/Masterminds/glide/vendor/github.com/codegangsta/cli/app.go:490 +0xc8
github.com/Masterminds/glide/vendor/github.com/codegangsta/cli.Command.Run(0x14bf842, 0x7, 0x14bde6b, 0x1, 0x0, 0x0, 0x0, 0x14cb14f, 0x20, 0x0, ...)
	/private/tmp/glide-20180907-75095-1ck8wl4/glide-0.13.1/src/github.com/Masterminds/glide/vendor/github.com/codegangsta/cli/command.go:210 +0x990
github.com/Masterminds/glide/vendor/github.com/codegangsta/cli.(*App).Run(0xc0001ea000, 0xc00009c090, 0x3, 0x3, 0x0, 0x0)
	/private/tmp/glide-20180907-75095-1ck8wl4/glide-0.13.1/src/github.com/Masterminds/glide/vendor/github.com/codegangsta/cli/app.go:255 +0x687
main.main()
	/private/tmp/glide-20180907-75095-1ck8wl4/glide-0.13.1/src/github.com/Masterminds/glide/glide.go:104 +0x556
make: *** [bootstrap] Error 2
```

**Comment by mattfarina** [comment]
Ick... just learned about this. I have a Mojave machine I'll go take a look at this on.

**Comment by fxcoudert** [comment]
As I stated above, with glide 0.13.1 we see this on all macOS versions. We previously believed it to be Mojave-only because we only tested it on Mojave as part of the beta testing, and because it had previously built fine on other macOS versions.

**Comment by bspradling** [comment]
Any workarounds for this right now? Running into this issue right now.

**Comment by bacongobbler** [comment]
Sorry to hear that! Feel free to install the "official" client releases for the time being while we fix the issue. Instructions on fetching helm can be found here: https://github.com/helm/helm/releases/tag/v2.11.0

[Using the install script](https://docs.helm.sh/using_helm/#from-script):

```
$ curl https://raw.githubusercontent.com/helm/helm/master/scripts/get > get_helm.sh
$ chmod 700 get_helm.sh
$ ./get_helm.sh
```

Yes, you can `curl https://raw.githubusercontent.com/helm/helm/master/scripts/get | bash` that if you want to live on the edge.

**Comment by bacongobbler** [comment]
looks like this has been fixed and v2.11.0 is now on Homebrew. Thanks @mattfarina!
